webapp/src/ts/constants.ts

```diff
@@ -26,7 +45,10 @@ export const POUCHDB_OPTIONS = {
         opts.headers.set(header, POUCHDB_OPTIONS.remote_headers[header]);
       });
       opts.credentials = 'same-origin';
-      return window.PouchDB.fetch(url, opts);
+      const isChangesRequest = parsedUrl.pathname.startsWith('/_changes');
+      return window.PouchDB
+        .fetch(url, opts)
+        .then(response => isChangesRequest ? processChangesResponse(response) : response);
```

Comment:
I'd feel a little more comfortable not adding the `.then` if we're not going to do anything, eg:

```suggestion
      const promise = window.PouchDB.fetch(url, opts);
      if (parsedUrl.pathname.startsWith('/_changes')) {
        promise.then(response => processChangesResponse(response));
      }
      return promise;
```